lib/opam_solve.ml

```diff
@@ -201,10 +221,15 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
           require_cross_compile
           && List.exists ~f:candidate_cross_compile candidates
         in
+        let preferred_version =
+          let open Option.O in
+          let* preferred_versions = preferred_versions in
```

Comment:
I feel like this would be nicer simpler if it was just am empty map, saves one indirection and expresses the same semantically.

Reply:
Yeah, I went for an option when I started this as I thought it would allow more optimisation to avoid running useless operations over and over when there's no preferred_versions but it was indeed not very bright as looking up an empty map is just as efficient as checking whether an option is `None` either way.